src/accounts.rs

```diff
@@ -87,22 +78,21 @@ impl AccountsDB {
         hash(&serialize(&ordered_accounts).unwrap())
     }
 
-    fn load(&self, pubkey: &Pubkey) -> Option<&Account> {
-        if let Some(account) = self.accounts.get(pubkey) {
-            return Some(account);
-        }
-
-        for (accounts, _) in &self.checkpoints {
-            if let Some(account) = accounts.get(pubkey) {
-                return Some(account);
+    fn load<U>(checkpoints: &[U], pubkey: &Pubkey) -> Option<Account>
+    where
+        U: Deref<Target = Self>,
+    {
+        for db in checkpoints {
+            if let Some(account) = db.accounts.get(pubkey) {
+                return Some(account.clone());
             }
         }
         None
     }
-
-    pub fn store(&mut self, pubkey: &Pubkey, account: &Account) {
+    /// purge == checkpoints.is_empty()
+    pub fn store(&mut self, purge: bool, pubkey: &Pubkey, account: &Account) {
         if account.tokens == 0 {
-            if self.checkpoints.is_empty() {
+            if purge {
                 // purge if balance is 0 and no checkpoints
```

Comment:
This comment says to purge if the balance is zero, where is balance factored into the decision to purge?

Reply:
balance check is on the line above.  I added docs to explain what purge does.